syntax = "proto3";

option go_package = "./pb";

service Monitor {
  // Sends a greeting
  rpc ReadOutput (ReadOutputRequest) returns (ReadOutputReply) {}
  rpc GetProcessStatus (GetProcessStatusRequest) returns (GetProcessStatusReply) {}
}

// The request message containing the user's name.
message ReadOutputRequest {
  string taskId = 1;
  int32 size = 2;
  int64 offset = 3;
}

// The response message containing the greetings
message ReadOutputReply {
  bytes data = 1;
  bool endOfFile = 2;
  int64 newOffset = 3;
}

message GetProcessStatusRequest {

}

message GetProcessStatusReply {
  int32 processCount = 1;
  int64 totalMemory = 2;
  int64 totalData = 3;
  int64 totalShared = 4;
  int64 totalResident = 5;
}
